### PR TITLE
Fix and improve nightmode (see description)

### PR DESCRIPTION
**Pull request by BenCat07**
This commit first of all fixes nightmode, then second of all makes it more customizeable, and the third point is that it fixes antiafk

image
image
image

Note: There is currently a bug where it will spam your console if you change settings, this is avoidable through some game interfering methods, so i am waiting for feedback from @TotallyNotElite on this matter.

### CODE REVIEW

File: src/hooks/visual/FrameStageNotify.cpp [hunk 1]
```diff
@@ -26,7 +37,7 @@ DEFINE_HOOKED_METHOD(FrameStageNotify, void, void *this_, ClientFrameStage_t sta
 
     PROF_SECTION(FrameStageNotify_TOTAL);
 
-    if (old_nightmode != *nightmode)
+    if (old_nightmode_gui != *nightmode_gui || old_nightmode_world != *nightmode_world || old_nightmode_world_color != *nightmode_world_color || old_nightmode_gui_color != *nightmode_gui_color)
```
Comment (TotallyNotElite) [inline src/hooks/visual/FrameStageNotify.cpp:40]: Learn to use a~~n~~ ~~egg~~ bool. And callbacks.

File: src/hooks/visual/FrameStageNotify.cpp [hunk 2]
```diff
@@ -42,20 +53,48 @@ DEFINE_HOOKED_METHOD(FrameStageNotify, void, void *this_, ClientFrameStage_t sta
...
+                {
                     pMaterial->ColorModulate(1.0f, 1.0f, 1.0f);
+                }
             }
         }
-        old_nightmode = *nightmode;
+        old_nightmode_gui         = *nightmode_gui;
+        old_nightmode_gui_color   = *nightmode_gui_color;
+        old_nightmode_world       = *nightmode_world;
+        old_nightmode_world_color = *nightmode_world_color;
```
Comment (TotallyNotElite) [inline src/hooks/visual/FrameStageNotify.cpp:97]: no.

File: src/hooks/visual/FrameStageNotify.cpp (hunk 1 above)
Comment (BenCat07) [reply src/hooks/visual/FrameStageNotify.cpp:40]: Callbacks don't work in this case, it needs to be done in this exact loop, But yeah i'll make it look nicer

File: src/hooks/visual/FrameStageNotify.cpp (hunk 2 above)
Comment (BenCat07) [reply src/hooks/visual/FrameStageNotify.cpp:97]: see above

**Pull request merged**